### PR TITLE
Test cases for DeviceAgentServceImpl.

### PR DESCRIPTION
**Pull request by menakaj**
## Automation tests
 - Unit tests 
   > Code Coverage : 39%
 - Integration tests
   > N/A

### CODE REVIEW

**Review by amalhub — APPROVED** [review]

**Review by harshanL — APPROVED** [review]

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,655 @@
...
+                "getDeviceAccessAuthorizationService")).toReturn(this.deviceAccessAuthorizationService);
+        PowerMockito.stub(PowerMockito.method(DeviceMgtAPIUtils.class,
+                "getAuthenticatedUser")).toReturn(AUTHENTICATED_USER);
+
+        Device testDevice = DeviceMgtAPITestHelper.generateDummyDevice(TEST_DEVICE_TYPE, TEST_DEVICE_IDENTIFIER);
+
+        Mockito.when(this.deviceManagementProviderService.getDevice(Mockito.any())).thenReturn(testDevice);
+        Mockito.when(this.deviceAccessAuthorizationService.isUserAuthorized(Mockito.any(DeviceIdentifier.class)))
+                .thenReturn(true);
+        Mockito.when(this.deviceManagementProviderService.modifyEnrollment(Mockito.any())).thenThrow(new DeviceManagementException());
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:355]: Seems to be violating 120 line limit. Some other places, have the same issue. Please fix them

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,655 @@
...
+        Response response = deviceAgentService.updateDevice(TEST_DEVICE_TYPE, TEST_DEVICE_IDENTIFIER, testDevice);
+        Assert.assertNotNull(response, "Response should not be null");
+        Assert.assertEquals(response.getStatus(), Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
+                "The response status should be 500");
+        Mockito.reset(this.deviceManagementProviderService);
+        Mockito.reset(this.deviceAccessAuthorizationService);
+    }
+    @Test(description = "Test updating device success scenario.")
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:363]: Please add a space betwenn test cases

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,655 @@
...
+                .thenReturn(eventStreamDefinitionDto);
+        Mockito.when(eventStreamDefinitionDto.getPayloadData()).thenReturn(new EventStreamAttributeDto[]{});
+        EventsPublisherService eventPublisherService = Mockito.mock(EventsPublisherServiceImpl.class,
+                Mockito.RETURNS_MOCKS);
+        PowerMockito.stub(PowerMockito.method(DeviceMgtAPIUtils.class, "getEventPublisherService")).toReturn
+                (eventPublisherService);
+
+        Map<String, Object> payload = new HashMap<>();
+        CacheImpl cache = Mockito.mock(CacheImpl.class);
+
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:484]: Remove the un-wanted lines. Please correct in other places as well

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,655 @@
...
+    private DeviceAgentService deviceAgentService;
+    private EventStreamAdminServiceStub eventStreamAdminServiceStub;
+    private PrivilegedCarbonContext privilegedCarbonContext;
+    private CarbonContext carbonContext;
+    private CacheManager cacheManager;
+    private DeviceAccessAuthorizationService deviceAccessAuthorizationService;
+    private static final String TEST_DEVICE_TYPE = "TEST-DEVICE-TYPE";
+    private static final String TEST_DEVICE_IDENTIFIER = "11222334455";
+    private static final String AUTHENTICATED_USER = "admin";
+    private static final String TENANT_DOMAIN = "carbon.super";
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:89]: Shall we use super-tenant domain name from MutiTenantConstants.

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,655 @@
...
+                Mockito.RETURNS_MOCKS);
+        this.privilegedCarbonContext = Mockito.mock(PrivilegedCarbonContext.class, Mockito.RETURNS_MOCKS);
+        this.carbonContext = Mockito.mock(CarbonContext.class, Mockito.RETURNS_MOCKS);
+        this.eventStreamAdminServiceStub = Mockito.mock(EventStreamAdminServiceStub.class, Mockito.RETURNS_MOCKS);
+        demoDevice = DeviceMgtAPITestHelper.generateDummyDevice(TEST_DEVICE_TYPE, TEST_DEVICE_IDENTIFIER);
+        this.cacheManager = Mockito.mock(CacheManager.class, Mockito.RETURNS_MOCKS);
+    }
+
+    @Test(description = "Test device Enrollment when the device is null")
+    public void testEnrollDeviceWithDeviceIsNULL() {
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:114]: Shall we use the correct convention when naming the test methods

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java (hunk 1 above)
Comment (menakaj) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:355]: This is fixed in the commit 
https://github.com/wso2/carbon-device-mgt/pull/1082/commits/1a995dde138fc934f14bf626ea0506061f04f39e

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java (hunk 2 above)
Comment (menakaj) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:363]: This is fixed in the commit 
https://github.com/wso2/carbon-device-mgt/pull/1082/commits/1a995dde138fc934f14bf626ea0506061f04f39e

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java (hunk 3 above)
Comment (menakaj) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:484]: This is fixed in the commit 
https://github.com/wso2/carbon-device-mgt/pull/1082/commits/1a995dde138fc934f14bf626ea0506061f04f39e

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java (hunk 4 above)
Comment (menakaj) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:89]: This is fixed in the commit 
https://github.com/wso2/carbon-device-mgt/pull/1082/commits/1a995dde138fc934f14bf626ea0506061f04f39e

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java (hunk 5 above)
Comment (menakaj) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java:114]: This is fixed in the commit 
https://github.com/wso2/carbon-device-mgt/pull/1082/commits/1a995dde138fc934f14bf626ea0506061f04f39e

**Pull request merged**